src/containers/FactoryLoader.tsx

```diff
@@ -215,13 +234,32 @@ export class FactoryLoaderContainer extends React.PureComponent<Props, State> {
     this.setState({ currentStep: LoadFactorySteps.APPLYING_DEVFILE });
     await delay();
 
-    let workspace: che.Workspace | null = null;
-    try {
-      workspace = await this.props.createWorkspaceFromDevfile(devfile, undefined, undefined, attrs) as any;
-    } catch (e) {
-      this.showAlert(`Failed to create a workspace. ${e}`);
+    let workspace: che.Workspace | undefined;
+    if (this.state.createPolicy === 'peruser') {
+      workspace = this.props.allWorkspaces.find(workspace => {
+        return WorkspaceStatus[workspace.status] === WorkspaceStatus.RUNNING &&
+          workspace.attributes && workspace.attributes.stackName === attrs.stackName;
+      });
+      if (!workspace) {
+        workspace = this.props.allWorkspaces.find(workspace => {
+          return workspace.attributes && workspace.attributes.stackName === attrs.stackName;
+        });
+      }
+    }
+    if (workspace && WorkspaceStatus[workspace.status] === WorkspaceStatus.RUNNING) {
+      this.props.setWorkspaceId(workspace.id);
+      this.setState({ currentStep: LoadFactorySteps.START_WORKSPACE });
+      await this.openIde();
       return;
     }
+    if (!workspace) {
+      try {
+        workspace = await this.props.createWorkspaceFromDevfile(devfile, undefined, undefined, attrs) as any;
```

Comment:
I seem casting result to type `any` is not necessary:

```suggestion
        workspace = await this.props.createWorkspaceFromDevfile(devfile, undefined, undefined, attrs);
```